clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1896,6 +1902,185 @@ private void recordRegionHybridConsumptionStats(
     }
   }
 
+  @Override
+  protected boolean isHybridFollower(PartitionConsumptionState partitionConsumptionState) {
+    return isHybridMode() && (isDaVinciClient || partitionConsumptionState.getLeaderFollowerState().equals(STANDBY));
+  }
+
+  /**
+   * Checks whether the lag is acceptable for hybrid stores
+   * <p>
+   * If the instance is a standby or DaVinciClient: Also check if <br>
+   * 1. the feature is enabled and is AA enabled: In non AA stores, consumer task is unable to read HB messages from RT
+   *           topic (though a test consumer can still read it), leading to standby replicas waiting for leader completion
+   *           state header indefinitely, so disabling it until that issue is resolved. <br>
+   * 2. first HB SOS is received and <br>
+   * 3. leaderCompleteStatus header has been received and <br>
+   * 4. leader was completed and <br>
+   * 5. the last update time was within the configured time interval
+   */
+  @Override
+  protected boolean checkAndLogIfLagIsAcceptableForHybridStore(
+      PartitionConsumptionState pcs,
+      long offsetLag,
+      long offsetThreshold,
+      boolean shouldLogLag,
+      boolean isOffsetBasedLag,
+      long latestConsumedProducerTimestamp) {
+    boolean isLagAcceptable = offsetLag <= offsetThreshold;
+
+    if (isLagAcceptable && isHybridFollower(pcs)) {
+      if (!(getServerConfig().isLeaderCompleteStateCheckEnabled() && this.isActiveActiveReplicationEnabled())) {
+        isLagAcceptable = true;
```

Comment:
This looks like a bug to me. If AA enable is false (non-AA store), we return lag acceptable immediately?

Reply:
but the codes do match the comment:

> 1. the feature is enabled and is AA enabled: In non AA stores, consumer task is unable to read HB messages from RT

So the HB based approach for follower completion report only work for AA stores?

Reply:
yes that's right. The issue with non AA stores needs to be further debugged and can be added as a new fix on top of this.